packages/valory/skills/decision_maker_abci/behaviours/storage_manager.py

```diff
@@ -226,23 +227,39 @@ def _get_tools(
         ):
             yield from self.wait_for_condition_with_sleep(step)
 
-    def _try_recover_policy(self) -> Optional[EGreedyPolicy]:
-        """Try to recover the policy from the policy store."""
+    def _read_tool_accuracy_local(self) -> Optional[str]:
+        """Reads the accuracy info from the local policy store file."""
         try:
             policy_path = self.params.store_path / POLICY_STORE
             with open(policy_path, "r") as f:
                 policy = f.read()
-                return EGreedyPolicy.deserialize(policy)
+                return policy
+
         except Exception as e:
             self.context.logger.warning(f"Could not recover the policy: {e}.")
             return None
 
+    def _try_recover_policy(self) -> Optional[EGreedyPolicy]:
+        """Try to recover the policy from the policy store."""
+        policy = self._read_tool_accuracy_local()
+        if policy:
+            # verify for updated_ts in policy, EGreedyPolicy accuracy_store now stores updated_ts
+            if policy and "updated_ts" not in policy:
+                # convert json str to dictionary
+                policy_data = json.loads(policy)
+                # add updated_ts to policy
+                policy_data["accuracy_store"]["updated_ts"] = 0
+                policy = json.dumps(policy_data)
+            return EGreedyPolicy.deserialize(policy)
+        else:
+            return None
```

Comment:
Too hacky, why not update the `EGreedyPolicyDecoder` instead?

Reply:
Updateed EGreedyPolicyDecoder.